core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
With respect to deprecating the FirstOrder rule, it does havea few notable differences from this rule. 

Specifically this rule works only for users who have accounts with the site, I'm not sure if its desired to support anonymous users with this rule. The first order rule has to do those lookups for anonymous users in a haphazard attempt to prevent every order from being a "first" order for anonymous users, though it can be bypassed with a different email. I suspect that a number of stores may override the first order promotion to make this logic more robust. 

I would be willing to deprecate the first order rule if there is consensus on the matter, though I think it might make the logic more complex.

Regarding the order check in that class, I don't understand why that check is there. Completed orders should not have new promotions applied, save manually via an admin or cs rep. If this order is completed, I expect the promotion to be ineligible. If there is an error in thinking for this, please let me know.